HyperlaneDevExDemo.md

```diff
@@ -0,0 +1,27 @@
+# Hyperlane DevEx Demo
```

Comment:
We've been spelling it "DevX", can this be changed? I can't rename the Discord channel, I know that's DevEx but its an outlier. 😄